---
title: "sandbox_lab14"
output: html_document
---
.---
title: "Data Visualization: `ggplot` part 4"
output:
  html_document:
    theme: spacelab
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

## Learning Goals
*At the end of this exercise, you will be able to:*    
1. Build heat maps using `ggplot.` 
2. Create Violin plots.
3. Customize labels on axes using `labs` and `themes`.  
4. Use `color`, `fill`, and `group` to customize plots and improve overall aesthetics.  


## Resources
- [ggplot2 cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
- [R for Data Science](https://r4ds.had.co.nz/)
- [R Cookbook](http://www.cookbook-r.com/)
- [`ggplot` themes](https://ggplot2.tidyverse.org/reference/ggtheme.html)
- [Rebecca Barter `ggplot` Tutorial](http://www.rebeccabarter.com/blog/2017-11-17-ggplot2_tutorial/)
- [R Color Brewer](http://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3)

## Install `RColorBrewer`, `paletteer`, and `ggthemes`
`RColorBrewer` and `paletteer` are helpful in selecting palettes of color that work with your data. Please install these now, we will work with them in today's lab.
```{r, eval=FALSE, echo=TRUE}
#install.packages("RColorBrewer")
#install.packages("paletteer")
#install.packages("ggthemes")
```

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(RColorBrewer)
library(paletteer)
library(janitor)
library(here)
```

```{r}
options(scipen=999) #cancels the use of scientific notation for the session
```

## Review
Now that you have been introduced to `ggplot`, let's review the plot types from last week and learn how to manipulate their aesthetics to better suit our needs. Aesthetics make a significant difference, but you can take it too far so remember that the goal is to produce clean plots that are not distracting.  

##Resources
- [ggplot2 cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
- [`ggplot` themes](https://ggplot2.tidyverse.org/reference/ggtheme.html)
- [Rebecca Barter `ggplot` Tutorial](http://www.rebeccabarter.com/blog/2017-11-17-ggplot2_tutorial/)

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(here)
library(janitor)
```

## Load the data
Let's revisit the mammal life history data to practice our ggplot skills. The [data](http://esapubs.org/archive/ecol/E084/093/) are from: *S. K. Morgan Ernest. 2003. Life history characteristics of placental non-volant mammals. Ecology 84:3402.*
```{r message=FALSE, warning=FALSE}
life_history <- read_csv(here("data", "mammal_lifehistories_v2.csv")) %>% na_if("-999") %>% clean_names()
```

```{r}
glimpse(life_history)
```

## Violin Plots
Violin plots show the distibution of contiuous data across a categorical variable.
Violin plots are similar to box plots and dot plots

lets get an idea of the data structure
```{r}
life_history %>% 
  count(order, sort=T)
```
```{r}
str(life_history)
```

Here, we visualize the distribution of average mass across orders. 
Since the orders are so diverse in size (an ant v.s. a giraffe), lets use a log function on mass. 
Also, let's flip the axes to make the graph more friendly for landscape orientation.
```{r}
life_history %>%
  ggplot(aes(x = order, y = log(mass))) +
  geom_boxplot(na.rm = TRUE)+
  coord_flip()
```


Let's focus in on a small set of primates, _Cercopithecidae_ family, i.e. Old-world monkeys, and generate a boxplot for each taxonomic genus.
```{r}
life_history %>%
  filter(family == "Cercopithecidae")%>%
  ggplot(aes(x = genus, y = mass)) +
  geom_boxplot(na.rm = TRUE) +
  coord_flip()
```

Box plots are great at showing median, skew, and outliers. 
We can visualize skew better by adding a point layer with geom_point()

```{r}
life_history %>%
    filter(family == "Cercopithecidae")%>%
    ggplot(aes(x = genus, y = mass)) +
    geom_boxplot(color = "grey", na.rm = TRUE) +
    geom_point(alpha = 0.5, color = "red", na.rm = TRUE) + #alpha indicates transparency 
    coord_flip()
```
By using the geom_point() layer, we see that there are only a few observations for each group (genus).
Lets further zoom in on 3 genera of monkey: Macaca, Cercopithecus, and Trachypithecus


we can use violin plots to visualize the distribution of these points on another axis:

```{r, warning=FALSE}
life_history %>%
  filter(genus == "Macaca" | genus == "Cercopithecus"| genus == "Trachypithecus")%>%
  ggplot(aes(x = genus, y = log(mass))) +
  geom_boxplot(color = "grey", na.rm = TRUE) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(alpha = .3, na.rm = TRUE) +
  coord_flip()
```



Lets zoom out and apply this graph style to a few primate families, Cercopithecidae, Hominidae and Cebidae 
```{r}
life_history %>%
  filter(family == "Cercopithecidae" | family == "Indridae" | family == "Cebidae") %>%
  ggplot(aes(x = family, y = mass)) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(alpha = .3, na.rm = TRUE) +
  coord_flip()
```
we can use violin plots to visualize the distribution of these points on another axis:

by default, the violin plot height is based on normalized area. we can switch to count with:

```{r}
life_history %>%
  filter(family == "Cercopithecidae" | family == "Indridae" | family == "Cebidae") %>%
  ggplot(aes(x = family, y = mass)) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(scale = "count", alpha = .3, na.rm = TRUE) +
  coord_flip()
```



## Heat maps
[heat maps](https://www.r-graph-gallery.com/79-levelplot-with-ggplot2.html) are a great tool for visualizing

### `pivot_longer()`  
Scientists frequently use spreadsheets that are organized to make data entry efficient. This is often referred to as **wide format**. Unfortunately, the wide format creates a problem because column names may actually represent values of a variable. The command `pivot_longer()` shifts data from wide to long format.   

Rules:  
+ `pivot_longer`(cols, names_to, values_to)
+ `cols` - Columns to pivot to longer format
+ `names_to` - Name of the new column; it will contain the column names of gathered columns as values
+ `values_to` - Name of the new column; it will contain the data stored in the values of gathered columns

## Example 1: column names are data
The following data show results from a drug trial with four treatments on six patients. The values represent resting heart rate.  
```{r}
heartrate <- readr::read_csv("data/heartrate.csv")
heartrate
```


```{r}
names(heartrate)
```

```{r}
heartrate_long <- heartrate %>% 
  pivot_longer(-patient, #patient does not pivot
               names_to = "drug", 
               values_to = "heartrate"
               )
heartrate_long
```
```{r}
heartrate_long %>%
  ggplot(aes(patient, drug, fill= heartrate)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```

lets try with data that considers time





Remember this large data set from the [United Nations Food and Agriculture Organization](http://www.fao.org/about/en/) on world fisheries? These data are pretty wild, so we need to do some cleaning. First, load the data.  

Load the data `FAO_1950to2012_111914.csv` as a new object titled `fisheries`.
```{r}
fisheries <- readr::read_csv(file = "data/FAO_1950to2012_111914.csv")
```




```{r}
fisheries <- janitor::clean_names(fisheries)
names(fisheries)
```

lets practice using across() to change our interesting varriables to factors
```{r}
fisheries <- 
  fisheries %>% mutate(across(c(country, isscaap_group_number, asfis_species_number, fao_major_fishing_area), as_factor))
```

Notice that these data are in wide format - year (a variable) is represented as columns. Let's practice pivoting long so we can make a hearmap: 
```{r}
fisheries_long <- fisheries %>% 
pivot_longer(cols = starts_with("x"),
               names_to = "year",
               values_to = "catch",
               values_drop_na = TRUE) %>% 
  mutate(year= as.numeric(str_replace(year, 'x', ''))) %>% 
  mutate(catch= str_replace(catch, c(' F'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('...'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('-'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('0 0'), replacement = ''))

fisheries_long$catch <- as.numeric(fisheries_long$catch)
```


which countries caught a lot of squid in the 90's?
```{r}

squid_tycoons <- fisheries_long %>%  
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  filter(between(year, 1990, 1999)) %>%
  group_by(country) %>% 
  summarize(catch_total=sum(catch, na.rm=T)) %>% 
  arrange(desc(catch_total))

squid_tycoons
```


```{r}

fish_tycoons <- fisheries_long %>%
  filter(between(year, 2000, 2006)) %>%
  group_by(country) %>% 
  summarize(catch_total=sum(catch, na.rm=T), catch = catch, year = year) %>% 
  arrange(desc(catch_total))

fish_tycoons%>%
   filter( country == "Japan" |			
   country == "Korea, Republic of" |
   country == "Taiwan Province of China" |
     country == "Argentina" |
   country == "China" |
   country == "Mexico"	|
   country == "Thailand" |
   country == "France" |
   country == "United States of America" |
   country == "Spain"
   )%>%
  ggplot(aes(x = year, y = country, fill= log(catch))) + 
  geom_tile(na.rm = T) +
  scale_fill_gradient(low="yellow", high="red")

```
  
```{r}
  

squid_tycoons %>%
   filter( country == "China" |			
   country == "Korea, Republic of" |
   country == "Japan" |
   country == "Taiwan Province of China" |
   country == "United States of America" |
   country == "Peru"	|
   country == "Argentina" |
   country == "Viet Nam" |
   country == "Thailand" |
   country == "Spain"
   )%>%
  ggplot(aes(x = year, y = country, fill= catch)) + 
  geom_tile(na.rm = T) +
  scale_fill_gradient(low="yellow", high="red")


```


```


%>%
  group_by(country, year, catch)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  summarise(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch)) %>%
  slice_max(tot_cephalopod_catch, n = 10)


```
Manipulated data:
```{r}
squid_champs <- fisheries_long %>%  
  group_by(country)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>%  
  select(country, year, catch)%>% 
  mutate(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch))

squid_champs
```

```{r}
squid_champs %>%
  ggplot(aes(x = year, y = country, fill= tot_cephalopod_catch)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```


```{r}
squid_champs %>%  
  group_by(country)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  mutate(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch)) %>%
  slice_max(tot_cephalopod_catch, n = 10) %>%
  select(country, year, tot_cephalopod_catch)%>%
  ggplot(aes(x = year, y = country, fill= tot_cephalopod_catch)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")


```



```{r}
fisheries_tidy_focused %>%
  filter(between(year, 2000, 2006)) %>%
#   filter( country == "China" |			
# country == "Korea, Republic of" |
# country == "Japan" |
# country == "Taiwan Province of China" |
# country == "United States of America" |
# country == "Peru"	|
# country == "Argentina" |
# country == "Viet Nam" |
# country == "Thailand"
#   ) %>%
  group_by(country, year)%>%
  summarise(catch_tot = sum(catch, na.rm =T)) %>% 
  ggplot(aes(x = year, y = country, fill= catch_tot)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```


--------------------------







```{r}

fish_tycoons <- fisheries_long %>%
  filter(between(year, 2000, 2006)) %>%
  group_by(country) %>% 
  summarize(catch_total=sum(catch, na.rm=T), catch = catch, year = year) %>% 
  arrange(desc(catch_total))

fish_tycoons%>%
   filter( country == "Japan" |			
   country == "Korea, Republic of" |
   country == "Taiwan Province of China" |
     country == "Argentina" |
   country == "China" |
   country == "Mexico"	|
   country == "Thailand" |
   country == "France" |
   country == "United States of America" |
   country == "Spain"
   )%>%
  ggplot(aes(x = year, y = country, fill= log(catch))) + 
  geom_tile(na.rm = T) +
  scale_fill_gradient(low="yellow", high="red")

```
  
```{r}
  

squid_tycoons %>%
   filter( country == "China" |			
   country == "Korea, Republic of" |
   country == "Japan" |
   country == "Taiwan Province of China" |
   country == "United States of America" |
   country == "Peru"	|
   country == "Argentina" |
   country == "Viet Nam" |
   country == "Thailand" |
   country == "Spain"
   )%>%
  ggplot(aes(x = year, y = country, fill= catch)) + 
  geom_tile(na.rm = T) +
  scale_fill_gradient(low="yellow", high="red")


```


```


%>%
  group_by(country, year, catch)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  summarise(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch)) %>%
  slice_max(tot_cephalopod_catch, n = 10)


```
Manipulated data:
```{r}
squid_champs <- fisheries_long %>%  
  group_by(country)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>%  
  select(country, year, catch)%>% 
  mutate(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch))

squid_champs
```

```{r}
squid_champs %>%
  ggplot(aes(x = year, y = country, fill= tot_cephalopod_catch)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```


```{r}
squid_champs %>%  
  group_by(country)%>%
  filter(between(year, 2000, 2006)) %>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  mutate(tot_cephalopod_catch = sum(catch, na.rm =T)) %>% 
  arrange(desc(tot_cephalopod_catch)) %>%
  slice_max(tot_cephalopod_catch, n = 10) %>%
  select(country, year, tot_cephalopod_catch)%>%
  ggplot(aes(x = year, y = country, fill= tot_cephalopod_catch)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")


```



```{r}
fisheries_tidy_focused %>%
  filter(between(year, 2000, 2006)) %>%
#   filter( country == "China" |			
# country == "Korea, Republic of" |
# country == "Japan" |
# country == "Taiwan Province of China" |
# country == "United States of America" |
# country == "Peru"	|
# country == "Argentina" |
# country == "Viet Nam" |
# country == "Thailand"
#   ) %>%
  group_by(country, year)%>%
  summarise(catch_tot = sum(catch, na.rm =T)) %>% 
  ggplot(aes(x = year, y = country, fill= catch_tot)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```


--------------------------











```{r}
beachbugs <- read_csv(here("data", "beachbugs_long.csv"))
beachbugs
```
## Practice




1. Make a barplot that shows counts of ectotherms and endotherms. Label the axes, provide a title, and fill by `thermoregulation`.
```{r}

```

2. Make a scatterplot that compares log10.mass and homerange. Adjust the size of points by mass. Label the axes and provide a title.
```{r}

```

## That's it! Take a break and I will see you on Zoom!  

-->[Home](https://jmledford3115.github.io/datascibiol/)






## Practice


## That's it! Take a break and I will see you on Zoom!  










---------------
presented in lab:

## Lab meeting goals
* solicit feed back & ideas on how to introduce these concepts to people with non-begginer familiarity with the topic
* solicit content ideas and general knowledge
  + what r markdown features do you know and love?
  + if you recall, what are things you wish you'd learned about earlier?

## Learning Goals
*At the end of this exercise, you will be able to:*    
1. Create Violin plots.
2. Build heat maps using `ggplot.` 


## Resources
- [ggplot2 cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
- [R for Data Science](https://r4ds.had.co.nz/)
- [R Cookbook](http://www.cookbook-r.com/)
- [`ggplot` themes](https://ggplot2.tidyverse.org/reference/ggtheme.html)
- [Rebecca Barter `ggplot` Tutorial](http://www.rebeccabarter.com/blog/2017-11-17-ggplot2_tutorial/)
- [R Color Brewer](http://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3)

## Install `RColorBrewer`, `paletteer`, and `ggthemes`
`RColorBrewer` and `paletteer` are helpful in selecting palettes of color that work with your data. 
```{r, eval=FALSE, echo=TRUE}
#install.packages("RColorBrewer")
#install.packages("paletteer")
#install.packages("ggthemes")
```

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(RColorBrewer)
library(paletteer)
library(janitor)
library(here)
```

```{r}
options(scipen=999) #cancels the use of scientific notation for the session
```

##Resources
- [ggplot2 cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)
- [`ggplot` themes](https://ggplot2.tidyverse.org/reference/ggtheme.html)
- [Rebecca Barter `ggplot` Tutorial](http://www.rebeccabarter.com/blog/2017-11-17-ggplot2_tutorial/)

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(here)
library(janitor)
```

### Load the data

Let's revisit the mammal life history data to practice our ggplot skills. The [data](http://esapubs.org/archive/ecol/E084/093/) are from: *S. K. Morgan Ernest. 2003. Life history characteristics of placental non-volant mammals. Ecology 84:3402.*

```{r message=FALSE, warning=FALSE}
life_history <- read_csv(here("data", "mammal_lifehistories_v2.csv")) %>% na_if("-999") %>% clean_names()
```


### Violin Plots
Violin plots show the distribution of continuous data across a categorical variable.
Violin plots are similar to box plots and dot plots

lets get an idea of the data structure
```{r}
life_history %>% 
  count(order, sort=T)
```
```{r}
str(life_history)
```

Here, we visualize the distribution of average mass across orders. 
Since the orders are so diverse in size (an ant v.s. a giraffe), lets use a log function on mass. 
Also, let's flip the axes to make the graph more friendly for landscape orientation.
```{r}
life_history %>%
  ggplot(aes(x = order, y = log(mass))) +
  geom_boxplot(na.rm = TRUE)+
  coord_flip()
```


Let's focus in on a small set of primates, _Cercopithecidae_ family, i.e. Old-world monkeys, and generate a boxplot for each taxonomic genus.
```{r}
life_history %>%
  filter(family == "Cercopithecidae")%>%
  ggplot(aes(x = genus, y = mass)) +
  geom_boxplot(na.rm = TRUE) +
  coord_flip()
```

Box plots are great at showing median, skew, and outliers. 
We can visualize skew better by adding a point layer with `geom_point()`

```{r}
life_history %>%
    filter(family == "Cercopithecidae")%>%
    ggplot(aes(x = genus, y = mass)) +
    geom_boxplot(color = "grey", na.rm = TRUE) +
    geom_point(alpha = 0.5, color = "red", na.rm = TRUE) + #alpha indicates transparency 
    coord_flip()
```
By using the geom_point() layer, we see that there are only a few observations for each group (genus).
Lets further zoom in on 3 genera of monkey: Macaca, Cercopithecus, and Trachypithecus


we can use violin plots to visualize the distribution of these points on another axis:

```{r, warning=FALSE}
life_history %>%
  filter(genus == "Macaca" | genus == "Cercopithecus"| genus == "Trachypithecus")%>%
  ggplot(aes(x = genus, y = log(mass))) +
  geom_boxplot(color = "grey", na.rm = TRUE) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(alpha = .3, na.rm = TRUE) +
  coord_flip()
```



Lets zoom out and apply this graph style to a few primate families, Cercopithecidae, Hominidae and Cebidae 
```{r}
life_history %>%
  filter(family == "Cercopithecidae" | family == "Indridae" | family == "Cebidae") %>%
  ggplot(aes(x = family, y = mass)) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(alpha = .3, na.rm = TRUE) +
  coord_flip()
```
we can use violin plots to visualize the distribution of these points on another axis:

by default, the violin plot height is based on normalized area. we can switch to count with:

```{r}
life_history %>%
  filter(family == "Cercopithecidae" | family == "Indridae" | family == "Cebidae") %>%
  ggplot(aes(x = family, y = mass)) +
  geom_point(alpha = 0.3, color = "red", na.rm = TRUE) + #alpha indicates transparency 
  geom_violin(scale = "count", alpha = .3, na.rm = TRUE) +
  coord_flip()
```



## Heat maps
[heat maps](https://www.r-graph-gallery.com/79-levelplot-with-ggplot2.html) are a great tool for visualizing data across 2 categorical variables.


## Example 1: column names are data
The following data show results from a drug trial with four treatments on six patients. The values represent resting heart rate.  
```{r}
heartrate <- readr::read_csv("data/heartrate.csv")
heartrate
```

Let's pivot this data, so there is no data stored in variables
```{r}
heartrate_long <- heartrate %>% 
  pivot_longer(-patient, #patient does not pivot
               names_to = "drug", 
               values_to = "heartrate"
               )
heartrate_long
```

Now we can use `geom_tile()` to represent the contents
```{r}
heartrate_long %>%
  ggplot(aes(patient, drug, fill= heartrate)) + 
  geom_tile() +
  scale_fill_gradient(low="yellow", high="red")
```



Lets try this on a larger data set. 
Remember this one from the [United Nations Food and Agriculture Organization](http://www.fao.org/about/en/) on world fisheries? These data are pretty wild, so we need to do some cleaning. First, load the data.  

Load the data `FAO_1950to2012_111914.csv` as a new object titled `fisheries`.
```{r}
fisheries <- readr::read_csv(file = "data/FAO_1950to2012_111914.csv")
```



clean up names, 
```{r}
fisheries <- janitor::clean_names(fisheries)
names(fisheries)
```

lets practice using across() to change some interesting variables to factors
```{r}
fisheries <- 
  fisheries %>% mutate(across(c(country, isscaap_group_number, asfis_species_number, fao_major_fishing_area), as_factor))
```

Notice that these data are in wide format - year (a variable) is represented as columns. Let's practice pivoting long so we can make a heatmap: 
```{r}
fisheries_long <- fisheries %>% 
pivot_longer(cols = starts_with("x"),
               names_to = "year",
               values_to = "catch",
               values_drop_na = TRUE) %>% 
  mutate(year= as.numeric(str_replace(year, 'x', ''))) %>% 
  mutate(catch= str_replace(catch, c(' F'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('...'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('-'), replacement = '')) %>% 
  mutate(catch= str_replace(catch, c('0 0'), replacement = ''))

fisheries_long$catch <- as.numeric(fisheries_long$catch)
```


which countries caught a lot of squid in the 90's?
```{r}

fisheries_long %>%  
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  filter(between(year, 1990, 1999)) %>%
  group_by(country) %>% 
  summarize(catch_total=sum(catch, na.rm=T)) %>% 
  arrange(desc(catch_total))
```

```{r}

fish_tycoons <-   fisheries_long%>%
  filter(isscaap_taxonomic_group == "Squids, cuttlefishes, octopuses") %>% 
  filter(between(year, 1990, 1999)) %>%
   filter( country == "Japan" |			
   country == "Korea, Republic of" |
   country == "Taiwan Province of China" |
  country == "Argentina" |
   country == "China" |
   country == "Mexico"	|
   country == "Thailand" |
   country == "France" |
   country == "United States of America" |
   country == "Spain"
            )%>%
  group_by(country) %>% 
  summarize(catch_total=sum(catch, na.rm=T), catch = catch, year = year) %>% 
  arrange(desc(catch_total))
  
```
```{r}
fish_tycoons%>%
  ggplot(aes(x = year, y = country, fill= log(catch))) + 
  geom_tile(na.rm = T) +
  scale_fill_gradient(low="yellow", high="red")

```


## Formatting reports: recapping markdown

check out this (source)[https://rmarkdown.rstudio.com/authoring_basics.html] for more

We know about code chunks

```{r}
names(heartrate_long)
```

but we can also use back ticks to put a small bit of code in `line with the text`

we can insert hyperlinks into out reports:
like (this)[https://en.wikipedia.org/wiki/Hyperlink]

color = speakers

---
title: "Data Visualization: fancy plots"
output:
  html_document:
    theme: spacelab
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

## Learning Goals
*At the end of this exercise, you will be able to:*    
1. Build a violin plot using.
2. Build heat maps using `ggplot.`




## Setup


### install packages
```{r}

###Hannah, delete polypath
#install.packages("ggpolypath")



#install.packages("ggVennDiagram")
```

### Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)

library(janitor)
library(here)
library(palmerpenguins)

#hannah delete ggpp
#library(ggpolypath)

library(ggVennDiagram)
```

```{r}
options(scipen=999) #cancels the use of scientific notation for the session
```

## Venn


### homerange 
```{r}
homerange <- read_csv(here("data", "Tamburelloetal_HomeRangeDatabase.csv"))

homerange %>%
#  summarize(n=n_distinct(taxon))
  distinct(taxon)

homerange %>%
  distinct(locomotion)

subset_homerange <- homerange %>%
  filter(taxon =="mammals" | taxon == "birds" | taxon == "turtles")
subset_homerange

```
### super heros venn
```{r}
superhero_info <- readr::read_csv("data/heroes_information.csv", na = c("", "-99", "-"))%>%
  clean_names()

evil_vec <- superhero_info %>%
  filter(alignment == "bad")%>%
  pull(name)

red_eye_vec <- superhero_info %>%
  filter(eye_color == "red")%>%
  pull(name)

male_vec <- superhero_info %>%
  filter(gender == "Male")%>%
  pull(name)

bald_vec <- superhero_info%>%
  filter(hair_color == "No Hair")%>%
  pull(name)
  

venn_diag_category_list_2 <- list(evil_vec, red_eye_vec)
ggVennDiagram(venn_diag_category_list_2, category.names = c("Evil", "Red Eyes"))

venn_diag_category_list_3 <- list(evil_vec, red_eye_vec, male_vec)
ggVennDiagram(venn_diag_category_list_3, category.names = c("Evil", "Red Eyes", "Male"))

venn_diag_category_list_4 <- list(evil_vec, red_eye_vec, male_vec, bald_vec)
ggVennDiagram(venn_diag_category_list_4, category.names = c("Evil", "Red Eyes", "Male", "Bald"))
```



#### cleaning up venns
[great info here](https://cran.r-project.org/web/packages/ggVennDiagram/vignettes/using-ggVennDiagram.html)
```{r}
feature_list <- list(evil_vec, red_eye_vec, male_vec)

#library(RColorBrewer)
#library(ggthemes)

ggVennDiagram(feature_list,
              category.names = c("Evil", "Red Eyes", "Male"),
              edge_size = 1
              )+
  scale_fill_distiller(palette = "GnBu")+
  scale_color_brewer(palette = "Greys") + 
  labs(title = "Four sets Venn Diagram",
         subtitle = "generated by `ggVennDiagram`",
         caption = ("you can add a caption here, like the date")
  )


###
#ggVennDiagram(feature_list,
#              edge_lty = "dashed",              # Changes border line-type
#              edge_size = 1,
#              category.names = c("Evil", "Red Eyes", "Male")
#              )+
#  scale_fill_distiller(palette = "RdBu") +      # Changes fill of circles
#  scale_color_brewer(palette = "BuPu") +        # Changes border color
  
```

```{r}
              category.names = c("Evil", "Red Eyes", "Male"), 
              show_intersect = TRUE)


ggVennDiagram(venn_diag_category_list_3, 
              category.names = c("Evil", "Red Eyes", "Male")+
              scale_fill_brewer(palette = "BuPu")+, 
              show_intersect = TRUE,
              fill)+
              guides(fill = "BuPu") +
#              color = "black", 
#              fill = "BuPu",
              theme(legend.title = element_text(color = "red"), legend.position = "bottom")
              
#  guides(fill = guide_legend(title = "num observations")) +
  theme(legend.title = element_text(color = "red"), legend.position = "bottom")

###



ggVennDiagram(venn_diag_category_list_3, category.names = c("Evil", "Red Eyes", "Male"), show_intersect = TRUE)



#my_pallete <- brewer.pal(5, GnBu)

#ggVennDiagram(venn_diag_category_list_3, color = "black", lwd = 0.8, lty = 1) + 
#  scale_fill_gradient(low = "#F4FAFE", high = "#4981BF")

ggVennDiagram(venn_diag_category_list_3, color = "black", fill = my_pallete) + 
#  guides(fill = guide_legend(title = "num observations")) +
  theme(legend.title = element_text(color = "red"), legend.position = "bottom")

```

So like, how do you make a venn diagram? what are they actually good for?
One useful place-- 2 factoral catagories with more than 2 factors each.

So looking at the hr data, there are 9 taxa. for simplicity, lets select 3. 

so among birds, mammals, and turtles, there is a mix of walking swimming and flying. 
how do the species in these categories fall into these locomotive cats?




### homerange ven

```{r}
homerange <- read_csv(here("data", "Tamburelloetal_HomeRangeDatabase.csv"))

#homerange %>%
#  summarize(n=n_distinct(taxon))
#  distinct(taxon)
#  distinct(locomotion)

#update home range:

head(homerange)
```
```{r}

homerange_ver2 <-  homerange %>%
  add_row(taxon = "mammal", common.name == "bat", locomotion == "flying", trophic.guild == "omnivore")



# are there bats in this data?
bird_vec <- homerange %>%
  filter(taxon =="birds")%>%
  pull(common.name)

flighted_vec <- homerange %>%
  filter(locomotion == "flying")%>%
  pull(common.name)

venn_diag_category_list <- list(fly = flighted_vec, bird = bird_vec)
ggVennDiagram(venn_diag_category_list, category.names = c("Flighted", "A Bird"))



```




```{r}
library(venn)



#venn(gene_list)




library(ggVennDiagram)

genes <- paste0("gene",1:1000)
set.seed(20210302)
gene_list <- list(A = sample(genes,100),
                  B = sample(genes,200),
                  C = sample(genes,300),
                  D = sample(genes,200))


p1 <- ggVennDiagram(gene_list, 
                    category.names = c("a very long name","short name","name","another name"))
p1
```


















### Violin Plots
Violin plots show the distribution of continuous data across a categorical variable.
Violin plots are a lot like a density plots, often displayed vertically. They are nice for comparing among groups, and are common in publications. 


### life_history
Let's revisit the mammal life history data to practice our ggplot skills. The [data](http://esapubs.org/archive/ecol/E084/093/) are from: *S. K. Morgan Ernest. 2003. Life history characteristics of placental non-volant mammals. Ecology 84:3402.*

```{r message=FALSE, warning=FALSE}
#life_history <- read_csv(here("data", "mammal_lifehistories_v2.csv")) %>% na_if("-999") %>% clean_names()
```






### life hist data
lets remind ourselves of the structure of the `life_history` data:

```{r}
#str(life_history)
```

```{r}
#head(life_history)
```



















-----------










```{r}
qpcr_untidy <- read_csv("data/qpcr_untidy.csv")
```


# Part 2 

We will also use this
[Homerange](http://datadryad.org/resource/doi:10.5061/dryad.q5j65/1). The data are from: Tamburello N, Cote IM, Dulvy NK (2015) Energy and the scaling of animal space use. The American Naturalist 186(2):196-211. http://dx.doi.org/10.1086/682070.  




